src/pymmcore_widgets/mda/_save_widget.py

```diff
@@ -0,0 +1,182 @@
+
+        # if it's a valid writer, just update the combo
+        if extension and writer in WRITERS:
+            self._update_combo_text(writer)
+
+        # if the extension is not valid, clear the extension and get it form the combo
+        elif name := self.save_name.text():
+            self.save_name.setText(name + WRITERS[self._writer_combo.currentText()][0])
+
+    def _get_extension_from_name(self, name: str) -> str:
+        """Get the extension of the selected writer."""
+        for ext in EXTENSIONS:
+            if name.endswith(ext):
+                return ext
+        return ""
+
+    def _get_writer_from_extension(self, extension: str) -> str:
+        """Get the writer from the extension."""
+        for writer, ext in WRITERS.items():
+            for e in ext:
+                if extension == e:
+                    return writer
+        raise ValueError(f"Extension {extension} not found in {WRITERS}")
+
+    def _update_combo_text(self, writer: str) -> None:
+        """Update the writer combo and trigger the text changed signal."""
+        # blocking and then manually calling _on_writer_combo_text_changed because
+        # if the text of the combo does not change, the currentTextChanged signal is not
+        # emitted and we need it to add the extension to the name
+        with signals_blocked(self._writer_combo):
+            self._writer_combo.setCurrentText(writer)
+        self._on_writer_combo_text_changed(writer)
+
+    def _on_writer_combo_text_changed(self, writer: str) -> None:
+        """Update the save name extension when the writer is changed."""
+        self.name_label.setText(SUBFOLDER if writer == "tiff-sequence" else FILE_NAME)
+        extension = self._get_extension_from_name(self.save_name.text())
+        if name := self._remove_extension(self.save_name.text()):
+            # get index of the extension in the list or get the first one
+            # this is useful in the case of the ome-tiff writer that has two extensions
+            # .ome.tiff and .ome.tif
+            idx = next(
+                (i for i, ext in enumerate(WRITERS[writer]) if extension == ext), 0
+            )
+            self.save_name.setText(name + WRITERS[writer][idx])
+
+    def _remove_extension(self, name: str) -> str:
+        """Remove the extension from the name if it's there."""
+        return next((name.replace(ext, "") for ext in EXTENSIONS if ext in name), name)
+
+    def value(self) -> str | None:
+        """Return current state of the save widget."""
+        if (
+            not self.isChecked()
+            or not self.save_dir.text()
+            or not self.save_name.text()
+        ):
+            return None
+
+        # returning a str because the MDASequence yaml does not support Path yet
+        # maybe we can add support for Path in the future
+        return str(Path(self.save_dir.text()) / self.save_name.text())
+
+    def setValue(self, value: Path | str) -> None:
```

Comment:
this method should also return to the ability to accept a dict, since it better matches the state of the widget.  It can *also* accept a string or path of you want, but that should just split it into the standard dict